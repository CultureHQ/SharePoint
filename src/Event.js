import React, { Fragment } from "react";

import formatTimestamp from "./format-timestamp";
import styles from "./styles";

import Cap from "./Cap";
import Metadata from "./Metadata";
import Points from "./Points";
import RSVPsPreview from "./RSVPsPreview";
import UserLink from "./UserLink";

import { PLATFORM_ROOT, PREVIEW_LIMIT } from "./config";

const Event = ({
  event: {
    id,
    name,
    image: { mediumUrl },
    sponsored,
    survey,
    cancelledAt,
    acceptedCount,
    rsvpPreview,
    startsAt,
    endsAt,
    host,
    location,
    cap
  }
}) => {
  const eventUrl = `${PLATFORM_ROOT}/events/${id}`;

  return (
    <div className={styles.event}>
      <div className={styles.eventHeader}>
        <div className={styles.eventLeft}>
          <a style={{ backgroundImage: `url(${mediumUrl})` }} href={eventUrl}>
            <Points sponsored={sponsored} survey={survey} />
            <span className={styles.eventName}>{name}</span>
          </a>
        </div>
        <Metadata
          cancelledAt={cancelledAt}
          startsAt={startsAt}
          endsAt={endsAt}
          location={location}
          host={host}
          acceptedCount={acceptedCount}
          cap={cap}
        />
      </div>
      <div className={styles.eventFooter}>
        <a className={styles.moreInfo} href={eventUrl}>More Info</a>
        <RSVPsPreview acceptedCount={acceptedCount} rsvpPreview={rsvpPreview} />
      </div>
    </div>
  );
};

export default Event;
