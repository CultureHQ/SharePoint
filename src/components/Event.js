import React, { Fragment } from "react";

import styles from "../styles";

import Cap from "./Cap";
import EventFooter from "./EventFooter";
import EventHeader from "./EventHeader";
import EventImage from "./EventImage";
import Metadata from "./Metadata";
import Points from "./Points";
import RSVPsPreview from "./RSVPsPreview";
import UserLink from "./UserLink";

const Event = ({
  event: {
    id,
    name,
    image: { mediumUrl },
    sponsored,
    survey,
    cancelledAt,
    acceptedCount,
    rsvpPreview,
    startsAt,
    endsAt,
    host,
    location,
    cap
  }
}) => (
  <div className={styles.event}>
    <EventHeader
      mediumUrl={mediumUrl}
      id={id}
      sponsored={sponsored}
      survey={survey}
      name={name}
      cancelledAt={cancelledAt}
      startsAt={startsAt}
      endsAt={endsAt}
      location={location}
      host={host}
      acceptedCount={acceptedCount}
      cap={cap}
    />
    <EventFooter
      id={id}
      acceptedCount={acceptedCount}
      rsvpPreview={rsvpPreview}
    />
  </div>
);

export default Event;
