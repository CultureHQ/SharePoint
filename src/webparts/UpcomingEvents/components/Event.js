import React from "react";

import styles from "../styles.css";

import EventImage from "./EventImage";
import EventMetadata from "./EventMetadata";
import Rsvps from "./Rsvps";

const Event = ({ event }) => (
  <div className={styles.event}>
    <div className={styles.eventHeader}>
      <EventImage event={event} />
      <EventMetadata event={event} />
    </div>
    <div className={styles.eventFooter}>
      <a className={styles.moreInfo} href={event.href}>More Info</a>
      <Rsvps event={event} />
    </div>
  </div>
);

export default Event;
