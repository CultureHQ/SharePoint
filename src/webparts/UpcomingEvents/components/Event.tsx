import * as React from "react";

import styles from "../styles.module.scss";

export interface IInterest {
  id: number;
  name: string;
}

export interface IDepartment {
  id: number;
  name: string;
  createdAt: string;
  updatedAt: string;
}

export interface ILocation {
  id: number;
  name: string;
  createdAt: string;
  updatedAt: string;
}

export interface IOrganizationValue {
  id: number;
  name: string;
  createdAt: string;
  updatedAt: string;
}

export interface IUserAvatar {
  thumbUrl: string;
  mediumUrl: string;
  largeUrl: string;
  isDefault: boolean;
  updatedAt: string;
  originalUrl: string;
}

export interface IRsvp {
  id: number;
  responseType: string;
  rating: number;
  receiveNotifications: boolean;
  attended: string;
  createdAt: string;
  updatedAt: string;
  user: IUser;
  inviter: IUser;
}

export interface IUser {
  id: number;
  name: string;
  email: string;
  culturehqAdmin: boolean;
  organizationAdmin: boolean;
  active: boolean;
  title: string;
  claimed: boolean;
  createdAt: string;
  updatedAt: string;
  points: number;
  departments: [IDepartment];
  interests: [IInterest];
  location: ILocation;
  avatar: IUserAvatar;
  tenureInDays: number;
}

export interface IEventImage {
  thumbUrl: string;
  mediumUrl: string;
  largeUrl: string;
  isDefault: boolean;
  updatedAt: string;
  originalUrl: string;
}

export interface IEvent {
  id: number;
  name: string;
  details: string;
  startsAt: string;
  endsAt: string;
  eventType: string;
  sponsored: boolean;
  createdAt: string;
  updatedAt: string;
  location: string;
  visibility: string;
  cap: number;
  inviteOnly: boolean;
  openInvites: boolean;
  acceptedCount: number;
  timezone: string;
  cancelledAt: string;
  commentCount: number;
  allLocations: boolean;
  host: IUser;
  organizationValues: [IOrganizationValue];
  locations: [ILocation];
  interests: [IInterest];
  rsvpPreview: [IRsvp];
  image: IEventImage;
  rsvpForUser: IRsvp;
}

export interface IEventProps {
  event: IEvent;
}

export default class Event extends React.Component<IEventProps, {}> {
  public render(): React.ReactElement<IEventProps> {
    return (
      <div className={styles.event}>
        <div className={styles.eventHeader}>

        </div>
        <div className={styles.eventFooter}>

        </div>
      </div>
    );
  }
}

// <a className={styles.moreInfo} href={event.href}>More Info</a>
// <EventImage event={event} />
// <EventMetadata event={event} />
// <Rsvps event={event} />
