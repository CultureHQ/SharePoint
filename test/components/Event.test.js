import React from "react";
import { shallow } from "enzyme";

import Event from "../../src/components/Event";

test("matches snapshot", () => {
  const event = { href: "https://platform.culturehq.com/events/1" };
  expect(shallow(<Event event={event} />)).toMatchSnapshot();
});
