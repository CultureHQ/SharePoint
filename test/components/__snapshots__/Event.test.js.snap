// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Event
    event={
      Object {
        "href": "https://platform.culturehq.com/events/1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="eventHeader"
        >
          <EventImage
            event={
              Object {
                "href": "https://platform.culturehq.com/events/1",
              }
            }
          />
          <EventMetadata
            event={
              Object {
                "href": "https://platform.culturehq.com/events/1",
              }
            }
          />
        </div>,
        <div
          className="eventFooter"
        >
          <a
            className="moreInfo"
            href="https://platform.culturehq.com/events/1"
          >
            More Info
          </a>
          <Rsvps
            event={
              Object {
                "href": "https://platform.culturehq.com/events/1",
              }
            }
          />
        </div>,
      ],
      "className": "event",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <EventImage
              event={
                Object {
                  "href": "https://platform.culturehq.com/events/1",
                }
              }
            />,
            <EventMetadata
              event={
                Object {
                  "href": "https://platform.culturehq.com/events/1",
                }
              }
            />,
          ],
          "className": "eventHeader",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "event": Object {
                "href": "https://platform.culturehq.com/events/1",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "event": Object {
                "href": "https://platform.culturehq.com/events/1",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="moreInfo"
              href="https://platform.culturehq.com/events/1"
            >
              More Info
            </a>,
            <Rsvps
              event={
                Object {
                  "href": "https://platform.culturehq.com/events/1",
                }
              }
            />,
          ],
          "className": "eventFooter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "More Info",
              "className": "moreInfo",
              "href": "https://platform.culturehq.com/events/1",
            },
            "ref": null,
            "rendered": "More Info",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "event": Object {
                "href": "https://platform.culturehq.com/events/1",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="eventHeader"
          >
            <EventImage
              event={
                Object {
                  "href": "https://platform.culturehq.com/events/1",
                }
              }
            />
            <EventMetadata
              event={
                Object {
                  "href": "https://platform.culturehq.com/events/1",
                }
              }
            />
          </div>,
          <div
            className="eventFooter"
          >
            <a
              className="moreInfo"
              href="https://platform.culturehq.com/events/1"
            >
              More Info
            </a>
            <Rsvps
              event={
                Object {
                  "href": "https://platform.culturehq.com/events/1",
                }
              }
            />
          </div>,
        ],
        "className": "event",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <EventImage
                event={
                  Object {
                    "href": "https://platform.culturehq.com/events/1",
                  }
                }
              />,
              <EventMetadata
                event={
                  Object {
                    "href": "https://platform.culturehq.com/events/1",
                  }
                }
              />,
            ],
            "className": "eventHeader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "event": Object {
                  "href": "https://platform.culturehq.com/events/1",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "event": Object {
                  "href": "https://platform.culturehq.com/events/1",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="moreInfo"
                href="https://platform.culturehq.com/events/1"
              >
                More Info
              </a>,
              <Rsvps
                event={
                  Object {
                    "href": "https://platform.culturehq.com/events/1",
                  }
                }
              />,
            ],
            "className": "eventFooter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "More Info",
                "className": "moreInfo",
                "href": "https://platform.culturehq.com/events/1",
              },
              "ref": null,
              "rendered": "More Info",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "event": Object {
                  "href": "https://platform.culturehq.com/events/1",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
